docs/deprecation.md

```diff
@@ -0,0 +1,66 @@
+# Deprecation Policy
+
+The goal of this policy is to reduce the impact of changes on users and developers of the project by providing
+clear guidelines and a well-defined process for deprecating functionalities. This policy applies to both features
+and API interfaces.
+
+This policy describes what Starlette should follow pre-1.0, and what Starlette will follow post-1.0.
+
+## Starlette Versions
+
+Starlette follows [Semantic Versioning](https://semver.org/), with some additional constraints.
+While we are pre-1.0, we treat any SemVer changes requiring a:
+
+- **major** bump as a change requiring a **minor** bump.
+- **minor** bump as a change requiring a **patch** bump.
+
+## Deprecation Types
+
+We'll consider two kinds of deprecations: **Python version** and **feature** deprecations.
+
+### Python Version Deprecation
+
+Starlette will aim to support a Python version until the [EOL date of that version](https://endoflife.date/python).
+When a Python version reaches EOL, Starlette will drop support for that version in the next minor release.
+
+!!! warning
+    Pre-1.0 Starlette will also drop support in the next minor release.
+
+The drop of Python version support will be documented in the release notes, but the user will **not** be warned it.
+
+### Feature Deprecation
+
+Starlette will deprecate a feature in the next minor release after the feature is marked as deprecated.
```

Comment:
This is the "when". Should the wording be changed? @adriangb

Reply:
I think this is saying when we will deprecate the feature, the question was when do we put in deprecation warnings. Can users expect deprecation warnings to show up at any time, only in minor releases or only in major releases?

Reply:
The act of deprecating a feature is adding a deprecation warning. At least, this is how I understand.

Reply:
How about:

```suggestion
Starlette will deprecate a feature in the next major release after the feature is marked as deprecated (via a deprecation warning). Features can be marked as deprecated in any minor release.
```

Reply:
I'll apply _something_ on that line.